src/main/kotlin/io/snyk/plugin/snykcode/core/HashContentUtils.kt

```diff
@@ -1,34 +1,24 @@
 package io.snyk.plugin.snykcode.core
 
 import ai.deepcode.javaclient.core.HashContentUtilsBase
-import com.intellij.openapi.fileEditor.impl.LoadTextUtil
 import com.intellij.openapi.util.Computable
-import com.intellij.psi.PsiFile
+import java.nio.file.Files
 
 class HashContentUtils private constructor() : HashContentUtilsBase(
     PDU.instance
 ) {
     override fun doGetFileContent(file: Any): String {
-        val psiFile = PDU.toPsiFile(file)
+        require(file is SnykCodeFile) { "File $file must be of type SnykCodeFile but is ${file.javaClass.name}" }
         return RunUtils.computeInReadActionInSmartMode(
-            psiFile.project,
-            Computable { getPsiFileText(psiFile) }
-        ) ?: ""
-    }
-
-    /** Should be run inside **Read action** !!!  */
-    private fun getPsiFileText(psiFile: PsiFile): String {
-        if (!psiFile.isValid) {
-            SCLogger.instance.logWarn("Invalid PsiFile: $psiFile")
-            return ""
-        }
-        // psiFile.getText() NOT works as it not synchronized to the source on disk
-        return try {
-            LoadTextUtil.loadText(psiFile.virtualFile).toString()
-        } catch (e: IllegalArgumentException) {
-            SCLogger.instance.logWarn(e.toString())
-            ""
-        }
+            file.project,
+            Computable {
+                return@Computable try {
+                    String(Files.readAllBytes(file.virtualFile.toNioPath()))
```

Comment:
loadText is not a good solution, I was running into errors when IntelliJ didn't understand the extension and presumed a file binary. That's why readBytes is better. I didn't see performance issues with DefinitelyTyped, else I'd have used a BufferedStream for sure. So I would keep it.

Reply:
DefinitelyTyped contains mostly small files. we need to check with project containing large source files.
Here the problems also could be with not synchronized content of file on disk and VF
If loadText() is not good I would look for another build in Idea solution to read file content, usually it will cover other corner cases for us too.

Reply:
I'm now using the underlying method that loadText also uses - VirtualFile#contentsToByteArray